### PR TITLE
Implementation of setinstancevar

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: https://github.com/rathena/rathena/issues/6353

* **Server Mode**: Both

* **Description of Pull Request**: 

Implementation of `setinstancevar` to simplify the data manipulation of instance variable outside the instance map.

### CODE REVIEW

**Comment by Everade** [comment]
Thanks a lot for your work, much appreciated!

Please don't forget about the array command as originaly suggested.
```
setinstancevar <instance_id>,<variable>,<value>;
setinstancearray <instance_id>,<array name>[<first value>],<value>{,<value>...<value>};

```
And renaming the command **getvariableofinstance** to **getinstancevar** would be great. That would help a lot.

**Comment by Atemo** [comment]
The system would require 'setinstancearray '+ more command like 'deleteinstancearray'.. to fully support the array manipulation. I don't plan to add them, at least not yet.
I can add an alias if you really want to
`BUILDIN_DEF2(getvariableofinstance, "getinstancevar","ri"),`
for me it's the same thing, as long as it does the job

**Comment by Lemongrass3110** [comment]
> I can add an alias if you really want to `BUILDIN_DEF2(getvariableofinstance, "getinstancevar","ri"),` for me it's the same thing, as long as it does the job

Maybe this would be good, because the new function has a new name schema. But maybe we can even rename the old from getvariableofinstance to getinstancevar and add it as BUILDIN_DEF2_DEPRECATED so people can change their scripts to the new name?

**Comment by Everade** [comment]
Lovely, thanks a bunch!

File: src/map/script.cpp [hunk 1]
```diff
@@ -26122,7 +26178,8 @@ struct script_function buildin_func[] = {
 	BUILDIN_DEF(camerainfo,"iii?"),
 
 	BUILDIN_DEF(achievement_condition,"i"),
-	BUILDIN_DEF(getvariableofinstance,"ri"),
+	BUILDIN_DEF(getinstancevar,"ri"),
+	BUILDIN_DEF2(getinstancevar, "getvariableofinstance","ri"),
```
Comment (Lemongrass3110) [inline src/map/script.cpp:26182]:
```suggestion
	BUILDIN_DEF2_DEPRECATED(getinstancevar, "getvariableofinstance","ri", "2021-12-05"),
```
Adjust to the date when you merge it. :-)

**Review by Lemongrass3110 — APPROVED** [review]
The references to getvariableofinstance in script_commands.txt need to be updated (rename of getvariableofinstance)

**Comment by Dieby** [comment]
image

Has anyone else had this error?

**Comment by Atemo** [comment]
Which error? I only see an explicit warning

**Comment by Dieby** [comment]
> Which error? I only see an explicit warning

Oh sorry, I explained myself wrong, I would like to know how to remove the warning

**Comment by vstumpf** [comment]
Use `getinstancevar` instead of `getvariableofinstance`

**Pull request merged**